### PR TITLE
Fixed compilation error in blast experimental gem

### PR DESCRIPTION
**Pull request by moraaar**
Fixes https://github.com/o3de/o3de/issues/15445

Fixed Blast unit tests as well. They pass successfully.

Signed-off-by: moraaar <moraaar@amazon.com>

### CODE REVIEW

**Comment by moraaar** [comment]
Hold reviews, Blast unit tests are not compiling, I'll push the fix soon.

**Review by hultonha — APPROVED** [review]
Thanks for jumping on this @moraaar, great work! 🥳

**Comment by hultonha** [comment]
> Hold reviews, Blast unit tests are not compiling, I'll push the fix soon.

Ah sorry just missed this, I'll take a look again soon when it's updated

**Review by SergeyAMZN — APPROVED** [review]

**Comment by moraaar** [comment]
Fixed blast unit tests. It turned out that was more complicated :D Please @SergeyAMZN @hultonha review now.

File: Gems/Blast/Code/Include/Blast/BlastActor.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace Blast
     class BlastActor
     {
     public:
-        AZ_CLASS_ALLOCATOR(BlastActor, AZ::SystemAllocator, 0);
+        AZ_CLASS_ALLOCATOR(BlastActor, AZ::SystemAllocator);
```
Comment (hultonha) [inline Gems/Blast/Code/Include/Blast/BlastActor.h:40]: Thanks for updating this too 🥳

**Review by hultonha — APPROVED** [review]
LGTM! 😄

**Review by lsemp3d — APPROVED** [review]

**Review by SergeyAMZN — APPROVED** [review]

**Pull request merged**